### ISSUE TITLE
warning: ‘template<class _Arg, class _Result> struct std::unary_function’ is deprecated

### ISSUE THREAD

**Issue by wkotlarski** [FlexibleSUSY/FlexibleSUSY#437]
Warning from g++ 12.1
```
In file included from src/slha_io.cpp:27:
slhaea/slhaea.h:1528:36: warning: ‘template<class _Arg, class _Result> struct std::unary_function’ is deprecated [-Wdeprecated-declarations]
 1528 |   struct key_matches : public std::unary_function<value_type, bool>
      |                                    ^~~~~~~~~~~~~~
In file included from /usr/include/c++/12/string:48,
                 from src/slha_format.hpp:22,
                 from src/slha_io.hpp:22,
                 from src/slha_io.cpp:19:
/usr/include/c++/12/bits/stl_function.h:117:12: note: declared here
  117 |     struct unary_function
      |            ^~~~~~~~~~~~~~
slhaea/slhaea.h:2354:36: warning: ‘template<class _Arg, class _Result> struct std::unary_function’ is deprecated [-Wdeprecated-declarations]
 2354 |   struct key_matches : public std::unary_function<value_type, bool>
      |                                    ^~~~~~~~~~~~~~
/usr/include/c++/12/bits/stl_function.h:117:12: note: declared here
  117 |     struct unary_function
      |            ^~~~~~~~~~~~~~
slhaea/slhaea.h:2375:46: warning: ‘template<class _Arg, class _Result> struct std::unary_function’ is deprecated [-Wdeprecated-declarations]
 2375 |   struct key_matches_block_def : public std::unary_function<value_type, bool>
      |                                              ^~~~~~~~~~~~~~
/usr/include/c++/12/bits/stl_function.h:117:12: note: declared here
  117 |     struct unary_function
      |            ^~~~~~~~~~~~~~
```

**Comment by Expander** [comment]
Thanks for finding this! Can you please file a bug report at https://github.com/fthomas/slhaea/issues ?

**Comment by pathron** [comment]
Is that still being maintained?  I thought the author left physics long ago?  Actually it looks like you doing the last commits but he did merge them in?

**Comment by Expander** [comment]
Yes, because @fthomas gave me write access to the repository, so I can maintain it.

**Comment by pathron** [comment]
That’s good.  Funnily enough at some point in the past there was some discussion in gambit of us taking it over and maintaining it (we also use it).  But looks like you have it covered :).

**Comment by Expander** [comment]
@wkotlarski Can you please check whether the issue is fixed in commit 68149f8?

**Comment by wkotlarski** [comment]
Yes, it is. Closing the issue.
